src/parseDef.ts

```diff
@@ -72,11 +72,12 @@ export type JsonSchema7Type = JsonSchema7TypeUnion & JsonSchema7Meta;
 
 export function parseDef(
   def: ZodTypeDef,
-  refs: Refs
+  refs: Refs,
+  isTop?: boolean
 ): JsonSchema7Type | undefined {
   const seenItem = refs.seen.find((x) => Object.is(x.def, def));
```

Comment:
Just something I noticed. Not sure you've been doing any performance benchmarks, but I've seen this pattern not scale well before. (`.find`/`.includes`/`.indexOf`/..., being called on a growing array in a nested loop).

Reply:
I guess we could just switch to a map